sklearn/calibration.py

```diff
@@ -334,7 +336,7 @@ def fit(self, X, y, sample_weight=None, **fit_params):
             # sample_weight checks
             fit_parameters = signature(base_estimator.fit).parameters
             supports_sw = "sample_weight" in fit_parameters
-            if sample_weight is not None and not supports_sw:
```

Comment:
I removed the condition `sample_weight is not None` here and in some other places because we are doing `sample_weight = _check_sample_weight(sample_weight, X)` in line 292 so sample_weight cannot be None anymore.